winterfell/services/claimService.js

```diff
@@ -2,18 +2,88 @@ var _ = require('lodash');
 var http = require('http-status-codes');
 var httpErrors = require('./../utils/httpErrors');
 var Claim = require('./../models/claim');
+var Form = require('./../models/form');
+var Q = require('q');
+var bulk = require('bulk-require');
+var formlyFields = bulk(__dirname + '/../forms/', ['*']);
+var expressions = require("angular-expressions");
 
 function ClaimService(app) {
   this.app = app;
 }
 
+/**
+ * For a given form and set of responses, calculate how many questions
+ * were answered (which is simply the number of keys in the responses),
+ * and more complicatedly: how many questions on the form were answerable,
+ * which requires evaluating the angular hideExpression attribute for
+ * each field against the current responses.
+ *
+ * @param formName
+ * @param data
+ * @returns {{answerable: number, answered: number}}
+ */
+function calculateProgress(formName, data) {
+  var evaluate, i;
+  var template = formlyFields[formName];
```

Comment:
That sounds good to me, slightly prefer formTitle over formName since its kind of long for a name..